File: lib/java/openlinktoken/src/main/java/org/openlinktoken/tokens/tokenizer/SHA3TokenDigest.java
/* SPDX-License-Identifier: MIT */
package org.openlinktoken.tokens.tokenizer;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

import org.openlinktoken.crypto.CryptoSuite;

/**
 * Calculates SHA3-256 token digests.
 */
public final class SHA3TokenDigest implements TokenDigest {

    /**
     * Calculates a SHA3-256 digest.
     *
     * @param value the token-signature bytes
     * @return the SHA3-256 digest
     * @throws NoSuchAlgorithmException if SHA3-256 is unavailable
     */
    @Override
    public byte[] digest(byte[] value) throws NoSuchAlgorithmException {
        return MessageDigest.getInstance(CryptoSuite.TOKEN_DIGEST_SHA3_256).digest(value);
    }
}

File: lib/python/openlinktoken/src/main/openlinktoken/tokens/tokenizer/sha3_token_digest.py
# SPDX-License-Identifier: MIT
"""SHA3-256 token digest implementation."""

import hashlib

from openlinktoken.tokens.tokenizer.token_digest import TokenDigest


class Sha3TokenDigest(TokenDigest):
    """Calculate SHA3-256 digests for token-signature bytes.

    Args:
        None.

    Returns:
        A ``Sha3TokenDigest`` instance.
    """

    def digest(self, value: bytes) -> bytes:
        """Return the SHA3-256 digest of supplied bytes.

        Args:
            value: Token-signature bytes to digest.

        Returns:
            The 32-byte SHA3-256 digest.
        """
        return hashlib.sha3_256(value).digest()
